core/src/main/scala/org/typelevel/keypool/KeyPool.scala

```diff
@@ -380,7 +385,10 @@ object KeyPool {
         kpVar <- Resource.make(
           Ref[F].of[PoolMap[A, (B, F[Unit])]](PoolMap.open(0, Map.empty[A, PoolList[(B, F[Unit])]]))
         )(kpVar => KeyPool.destroy(kpVar))
-        kpMaxTotalSem <- Resource.eval(Semaphore[F](kpMaxTotal.toLong))
+        kpMaxTotalSem <- fairness match {
+          case Fairness.Fifo => Resource.eval(RequestSemaphore[F](Fairness.Fifo, kpMaxTotal))
+          case Fairness.Lifo => Resource.eval(RequestSemaphore[F](Fairness.Lifo, kpMaxTotal))
+        }
```

Comment:
I assume this is an artifact of a slightly different prior design, but I don't think you need a pattern match for this

```suggestion
        kpMaxTotalSem <- Resource.eval(RequestSemaphore[F](fairness, kpMaxTotal))
```

Reply:
I'm even more sure now that I see the deprecated builder already has this simplification